src/app/modules/item/pages/chapter-user-progress/chapter-user-progress.component.html

```diff
@@ -47,22 +48,62 @@
           <td *ngFor="let col of columns">
             <ng-container [ngSwitch]="col.field">
               <ng-container *ngSwitchCase="'title'">
-                <a
-                  class="alg-link"
-                  [ngClass]="{'disabled': rowData.id === itemData?.item?.id}"
-                  (click)="onClick(rowData)"
-                >
-                  {{ rowData.title }}
-                </a>
+                <div class="title" *ngIf="itemData && itemData.currentResult?.attemptId as parentAttemptId">
+                  <a
+                    class="alg-link"
+                    [ngClass]="{'disabled': rowData.id === itemData.item.id}"
+                    [routerLink]="{
+                      id: rowData.id,
+                      contentType: rowData | contentTypeFromItem,
+                      path: itemData.route.path.concat([ itemData.item.id ]),
+                      parentAttemptId: parentAttemptId
+                    } | url : [ 'progress', 'chapter-user-progress' ]"
+                  >
+                    {{ rowData.title }}
+                  </a>
+
+                  <ng-container *ngIf="rowData.type === 'Task'">
+                    <a
+                      pButton
+                      class="alg-button p-button-rounded p-small-button p-button-outlined load-answer-button"
+                      [label]="(!!watchedGroup ? 'watchingGroup' : '') | i18nSelect : {
+                        watchingGroup: 'View the best answer',
+                        other: 'Reload the best answer'
+                      }"
+                      [routerLink]="{
+                        id: rowData.id,
+                        contentType: rowData | contentTypeFromItem,
+                        path: itemData.item.id === rowData.id ? itemData.route.path : itemData.route.path.concat([ itemData.item.id ]),
+                        parentAttemptId: parentAttemptId,
+                        answer: {
+                          best: true, participantId: watchedGroup ? watchedGroup.route.id : undefined
+                        }
+                      } | url"
+                      *ngIf="!watchedGroup || watchedGroup.route.isUser && rowData.submissions > 0 && (rowData.currentUserPermissions | allowsWatchingAnswers)"
```

Comment:
The "submissions" condition is on the whole thing as that's the same for the user itself (not watching): there is no point loading the best answer if there are no answer.
So it should look like:

```suggestion
                      *ngIf="rowData.submissions > 0 && (!watchedGroup || (watchedGroup.route.isUser && (rowData.currentUserPermissions | allowsWatchingAnswers)))"
```